### PR TITLE
deprecate project id from poi table, track deployment in metadata

### LINKED ISSUE #1662: Remove projectId from _poi table
The poi table currently has a column for the `projectId` but in every row of the table this will be the same value.

We should remove this column from the table and instead we can rely on a value in the _metadata table. This will require a migration to the table to ensure compatibility.

### PR DESCRIPTION
**Pull request by jiqiang90**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

TODO:

- [x] should not use Record type for deployments, user could switch back to same deployment and block height could be override

Fixes #1662 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] I have tested locally
- [ ] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] My code is up to date with the base branch

### CODE REVIEW

File: packages/node-core/src/indexer/project.service.ts [hunk 1]
```diff
@@ -195,6 +196,28 @@ export abstract class BaseProjectService<DS extends {startBlock?: number}> imple
       metadata.set('specName', specName);
     }
 
+    if (!existing.deployments) {
```
Comment (stwiname) [inline packages/node-core/src/indexer/project.service.ts:199]: I think this can be moved into a function

File: packages/utils/src/query/types.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type MetaData = {
   queryNodeVersion: string;
   startHeight?: number;
   rowCountEstimate: [TableEstimate];
+  deployments: Record<string, number>;
```
Comment (stwiname) [inline packages/utils/src/query/types.ts:21]: Do we need to make this queryable?

File: packages/node-core/src/indexer/entities/Poi.entity.ts [hunk 3]
```diff
@@ -10,16 +10,16 @@ export interface ProofOfIndex {
...
 
 export interface PoiModel extends Model<ProofOfIndex>, ProofOfIndex {}
 
 export type PoiRepo = typeof Model & {
   new (values?: unknown, options?: BuildOptions): PoiModel;
 };
-
-export function PoiFactory(sequelize: Sequelize, schema: string): PoiRepo {
+// Project Id is deprecated
+export function PoiFactoryDeprecate(sequelize: Sequelize, schema: string): PoiRepo {
```
Comment (stwiname) [inline packages/node-core/src/indexer/entities/Poi.entity.ts:22]: I don't think we should keep a deprecated version. We should drop the column instead

File: packages/utils/src/query/types.ts (hunk 2 above)
Comment (jiqiang90) [reply packages/utils/src/query/types.ts:21]: let's make it queryable, useful user track deployment history

**Review by stwiname — APPROVED** [review]

**Pull request merged**